### ISSUE TITLE
Bio.Phylo: "neighbor joining" tree construction fails for minimal alignment

### ISSUE THREAD

**Issue by hendrikweisser** [biopython/biopython#1323]
The "neighbor joining" algorithm for the construction of phylogenetic trees, implemented in `Bio.Phylo.TreeConstruction.DistanceTreeConstructor.nj` fails with the following error, if applied to a minimal sequence alignment containing only two sequences:
```
  File "...\Bio\Phylo\TreeConstruction.py", line 703, in nj
    if clades[0] == inner_clade:
UnboundLocalError: local variable 'inner_clade' referenced before assignment
```
This is with BioPython version 1.68, but looking at the code, the cause of the problem seems to be still present in the "master" branch: Following from [here (1.68)](https://github.com/biopython/biopython/blob/biopython-168/Bio/Phylo/TreeConstruction.py#L656) or [here ("master")](https://github.com/biopython/biopython/blob/master/Bio/Phylo/TreeConstruction.py#L694), the code that sets the `inner_clade` variable is only executed for distance matrices of size > 2.

A simple check could be added to avoid the error. However, since the UPGMA algorithm is able to successfully generate a (trivial) tree for such a minimal alignment, it would be great if the NJ algorithm could be adapted to do the same.

**Comment by peterjc** [comment]
CC @lijax, @etal

**Comment by peterjc** [comment]
This should be closed via #1891.
